addons/mail/static/src/discuss/web/avatar_card/avatar_card_popover.xml

```diff
@@ -14,7 +14,35 @@
                         </span>
                     </span>
                     <div class="d-flex flex-column o_card_user_infos overflow-hidden">
-                        <span class="o-mail-avatar-card-name fw-bold" t-if="name" t-esc="name"/>
+                        <div class="o-mail-avatar-card-name-actions d-flex align-items-center justify-content-center">
+                            <span class="o-mail-avatar-card-name fw-bold" t-if="name" t-esc="name"/>
+                            <div class="flex-grow-1"/>
+                            <t t-set="currentChannelRole" t-value="props.channelMember?.channel_id?.channel?.self_member_id?.channel_role"/>
+                            <Dropdown t-if="this.store.self_user?.is_admin || ['owner', 'admin'].includes(currentChannelRole)" position="'right-start'">
+                                <button class="btn btn-sm">
+                                    <i class="oi oi-ellipsis-v"/>
+                                </button>
+                                <t t-set-slot="content">
+                                    <t t-if="this.store.self_user?.is_admin or currentChannelRole == 'owner'">
+                                        <DropdownItem t-if="props.channelMember?.channel_role !== 'admin'" onSelected.bind="() => this.setChannelRole('admin')">
+                                            Set Channel Admin
+                                        </DropdownItem>
+                                        <DropdownItem t-if="props.channelMember?.channel_role === 'admin'" onSelected.bind="() => this.setChannelRole(false)">
+                                            Remove Channel Admin
+                                        </DropdownItem>
+                                        <DropdownItem t-if="props.channelMember?.channel_role !== 'owner'" onSelected.bind="() => this.setChannelRole('owner')">
+                                            Set Channel Owner
+                                        </DropdownItem>
+                                        <DropdownItem t-if="props.channelMember?.channel_role === 'owner'" onSelected.bind="() => this.setChannelRole(false)">
+                                            Remove Channel Owner
+                                        </DropdownItem>
```

Comment:
We should probably have the red text too for set admin/remove owner from self (as non-db admin)